rllib/core/learner/learner.py

```diff
@@ -1054,30 +1070,58 @@ def additional_update(
-                    main = self.module[module_id].main
-                    target = self.module[module_id].target
-                    for param, target_param in zip(
-                        main.parameters(), target.parameters()
-                    ):
-                        target_param.data.copy_(
-                            tau * param.data + (1.0 - tau) * target_param.data
-                        )
-
-        And inside a training loop:
-
-        .. code-block:: python
-
-            for _ in range(100):
-                sample = ...
-                self.learner.update(sample)
-                if self.learner.global_step % 10 == 0:
-                    self.learner.additional_update(tau=0.01)
+        .. doctest::
+
+            >>> from ray.rllib.algorithms.ppo.torch.ppo_torch_rl_module import (
+            ...     PPOTorchRLModule
+            ... )
+            >>> from ray.rllib.algorithms.ppo.ppo_catalog import PPOCatalog
+            >>> from ray.rllib.algorithms.ppo.torch.ppo_torch_learner import (
+            ...     PPOTorchLearner
+            ... )
+            >>> import gymnasium as gym
+
+            >>> env = gym.make("CartPole-v1")
+
+            Create a single agent RL module spec.
+            >>> module_spec = SingleAgentRLModuleSpec(
+            ...     module_class=PPOTorchRLModule,
+            ...     observation_space=env.observation_space,
+            ...     action_space=env.action_space,
+            ...     model_config_dict = {"hidden": [128, 128]},
+            ...     catalog_class = PPOCatalog,
+            ... )
+
+            >>> class CustomPPOLearner(PPOTorchLearner):
+            ...     def additional_update_for_module(self,
+            ...         *,
+            ...         module_id,
+            ...         hps,
+            ...         timestep,
+            ...         sampled_kl_values,
+            ...     ):
+            ...         tau = 0.005 # polyak averaging update parameter
+            ...         # perform polyak averaging update
+            ...         main = self.module[module_id].main
+            ...         target = self.module[module_id].target
+            ...         for param, target_param in zip(
+            ...             main.parameters(), target.parameters()
+            ...         ):
+            ...             target_param.data.copy_(
+            ...                 tau * param.data + (1.0 - tau) * target_param.data
+            ...             )
+
+            >>> learner = CustomPPOLearner(module_spec=module_spec)
+
+            Note: the learner should be built before it can be used.
+            >>> learner.build()
+
+            Inside a training loop, we can now call the additional update as we like:
+            >>> for i in range(100):
+            ...     sample = ... # doctest: +SKIP
+            ...     learner.update(sample) # doctest: +SKIP
+            ...     if i % 10 == 0:
+            ...         learner.additional_update()
```

Comment:
What would happen if set unskipped these and replaced the placeholder values?

Reply:
Same case as above, the sample is a batch that is just pretty hard to construct and takes up lots of space.